api/client/client.go

```diff
@@ -266,3 +252,39 @@ func NewAuthServer(addr string, provider auth.Provider) *AuthRemote {
 func (ar *AuthRemote) Sign(req []byte) ([]byte, error) {
 	return ar.AuthSign(req, nil, ar.provider)
 }
+
+// nomalizeURL checks for http/https protocol, appends "http" as default protocol if not defiend in url
+func normalizeURL(addr string) (*url.URL, error) {
+	addr = strings.TrimSpace(addr)
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		host, port, err = net.SplitHostPort(addr + ":8888")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if port != "" {
+		_, err = strconv.Atoi(port)
+		if err != nil {
+			return nil, err
+		}
+	}
+	addr = net.JoinHostPort(host, port)
+
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Opaque != "" {
+		u.Host = net.JoinHostPort(u.Scheme, u.Opaque)
+		u.Opaque = ""
+	}
+
+	if u.Scheme != "http" {
+		u.Scheme = "http"
```

Comment:
I think you missed an `s` here if we want to default to `https`.

This logic currently rewrites `https://` URLs to `http://` which is not correct.